minidump-processor/src/stackwalker/x86.rs

```diff
@@ -153,14 +153,23 @@ fn get_caller_by_scan(
             //    (after in memory) address_of_ip.
             let mut caller_bp = None;
 
-            if let Some(last_bp) = last_bp {
-                let address_of_bp = address_of_ip - POINTER_WIDTH;
-                if last_bp == address_of_bp {
-                    let bp = stack_memory.get_memory_at_address(address_of_bp as u64)?;
-                    if bp > address_of_ip {
-                        caller_bp = Some(bp);
-                    }
-                } else if last_bp >= address_of_ip + POINTER_WIDTH {
+            // TODO: the breakpad x86 analysis here varies slightly from
+            // the x64 logic, with no clear indication why. Also, the x86
+            // tests will complain if the x64 logic is used. Why?!
```

Comment:
The only thing I can offer is that the Breakpad stack walking code is built of years of scar tissue from dealing with real-world crash reports from Chrome and Firefox, and x86 is such a mess that it breaks in fantastical ways. Specifically, I think the rampant use of `-fomit-frame-pointer` in libraries where we didn't have symbols made the stack scanning code very important for getting useful stacks, but also very fiddly and full of special cases.

Given the relative usage of x86-64 and x86 nowadays it's probably fine to not port over every weird quirk from the Breakpad implementation if it makes your life harder, as long as it doesn't produce worse results for crashes you have to deal with. :)